include/serveropts.h

```diff
@@ -20,6 +21,17 @@ struct ConfigOpts
     int log_trunc_to_size;
     std::vector<std::string> graftlet_dirs;
     int lru_timeout_ms;
+
+    void check_asserts() const
```

Comment:
I'd suggest do not mention in function name some specific for language means. The name should reflect high level thing/goal and do not refer to implementation details. And in this case an usage of asserts is exactly implementation detail. We can check it with 'if'. After that we will change name of function? And its calls in code?

It should be rather something similar to check_correctness or check_consistency or validate_parameters/values.

Probably I'd do one step further. 
If we are checking validness passed in params then it should be done completely. I mean http address is bad not only when it's empty. We can check correctness in more smart way. Use some regex for example. Or it's overhead? 

As for timeouts.
Are they correct when its value is 1000? 1000 000? 1000 000 000? Do we expect such values?

Reply:
In this case the goal is to run asserts, exactly. It is convenient to assemble them together in this case. Checking that parameters are consistent should be done in other places.